### ISSUE TITLE
[Vaadin 19] Web Component generated by WebComponentExporter embedded in external SPA app- problem when vaadin Dialog is opened

### ISSUE THREAD

**Issue by darmro** [vaadin/flow#10300]
### Description of the problem
We have a Vaadin app that also delivers some web components generated via WebComponentExporter. These web components are embedded in a CMS system which is a SPA app. This CMS has a lot of its own pages, on few of them there are embedded our vaadin components. The user of course works in CMS by clicking on its menu items, clicking in the browser back button etc. 
Each CMS page has a content area (changeable) and some static page area (like each SPA has).

**In this scenario there is a big problem in situation when WebComponentExporter component openes a vaadin Dialog:**
1) web component is embeded in changeable content SPA area.
2) vaadin Dialog overlays (divs) are added to the bottom of the CMS page, so in its static area.
So when the user opens a Vaadin Dialog in our Vaadin component (embedded) and then clicks in the browser back button (thinking it will cause the window do close) the effect is that **SPA page is changing, but vaadin dialog remains open** (dialog from the previous page). Of course the user can then click cross (close) button in our dialog, but he will be confused.

To solve this I had to inject from web component to CMS page a javascript listener for popstate event, which calls vaadin app component (on server side) method, to find and close all opened window. But it is only a workaround of the problem.

If the vaadin Dialog overlay (all dialog divs) were added to the external page not on its bottom but in the same html container where vaadin web component is embedded it would solve the problem (for most cases). Such a Vaadin enhancement for developers using WebComponentExporter :)  But of course I don't know what other problems such behaviour could cause. 

### Versions:
    - Vaadin / Flow version: 19.0.1
    - Java version: 1.8

**Comment by pleku** [comment]
Hello. Did you try adding the dialog inside the embedded Flow web component with for example `someComponnet.add(dialog);` before calling `dialog.open();` ? I think this was the `vaadin-dialog` ends up elsewhere than as a child of the `UI` and I would expect that once the dialog is removed from the dom, it would remove the overlay too.

Another thing would be to make sure the `Dialog` itself handles detaching properly in this case, which might mean there is a problem with `WebComponentUI`.

**Comment by darmro** [comment]
Hi @pleku,
I've just just tried your idea with `someComponent.add(dialog)` before opening it, ie. I added the dialog to the component, which is then exporten by WebComponentExporter. This dialog is opened on button which is also in this component (web component).
Unfortunately nothing changed- the dialog overlay is still being placed in the bottom of the <body> of the page which embedds the web component, so the problem in the case when such a web component is embedded in the SPA app remains:
![image](https://user-images.githubusercontent.com/56316289/112700745-18a70a80-8e8f-11eb-85ca-3e175b12e6a1.png)

**Comment by darmro** [comment]
I think https://github.com/vaadin/flow/issues/10132 is related with the same problem- the Dialog overlay is outside the web component.

**Comment by pleku** [comment]
Yes, the dialog web component implementation is a source of a lot of problems with the overlay+transportation of content, but here I think it makes things broken in another way. While it should be quite trivial to reproduce this manually, an app which reproduces the issue helps us getting started on this. 😉
